.github/workflows/markdown-links-check.yml

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A workflow to check if PR got broken hyperlinks
+name: Check Markdown links
+
+on:
+  pull_request_target:
```

Comment:
`pull_request_target` -> `pull_request` to checkout the "refs/pull/:prNumber/merge"
otherwise the 
`- uses: actions/checkout@master`
will check out the BASE ref here (branch-22.04)

FYI we use pull_request_target in signoff-check is only because that the check is based on BASE branch instead of relying on developer's feature branch

Reply:
thanks and good to know that, one question is that do we need a GITHUB_TOKEN when we use `pull_request `?
If we don't add a GITHUB_TOKEN, will we hit `Error: Parameter token or opts.auth is required`?

Reply:
GITHUB_TOKEN is the default token of repo and is maintained by github action itself as ENV.

> If we don't add a GITHUB_TOKEN, will we hit Error: Parameter token or opts.auth is required?

For actions/checkout, no. 
I am not sure where you did get this error, if the some specific action require adding specific arg, then you can try add 
```
      with:
        token: ${{ secrets.GITHUB_TOKEN }}
```
above was just a example, this really depends on the action provider, if they require extra tokens (or access), they should have documented somewhere in their repo

Reply:
![image](https://user-images.githubusercontent.com/8086184/158517185-d948f160-b2e3-41a9-8f0a-c5ec4ca43259.png)
after changing to pull_request, you can see result in this PR's check status